lumicks/pylake/fdcurve.py

```diff
@@ -105,7 +105,9 @@ def subtract_distance_offset(self, distance_offset):
         distance_offset : float
         """
         new_curve = self.__copy__()
-        new_curve._distance_cache = Slice(TimeSeries(self.d.data - distance_offset, self.d.timestamps), self.d.labels)
+        new_distance = np.copy(self.d.data)
+        new_distance[new_distance > 0] = new_distance[new_distance > 0] - distance_offset
```

Comment:
On the special case in the docstring, definitely!

As for the negative distances, that's a good point. Yes it would. For curve alignment, I wouldn't expect this to happen (except possibly very sporadically because of noise), but it does result in a bit of a conundrum.

I guess we have a number of options:
- A. Clamp to zero (I really don't like this, since it will cause a statistical bias if this happens).
- B. Raise an exception when an offset is used that is larger than the minimum nonzero distance value found, cross fingers that it never fires because then ensemble work is impossible for that dataset.
- C. Just let it go negative. It's a bit tricky that the distance for failed tracking has been defined at zero, since technically you could land on zero when subtracting something (though I guess the chances are almost infinitely small).

Perhaps initially, an exception (B) is in order? Then we at least see if it's problem. It's also possible to just make this function private, so that people can't feed it bad things. What do you think?

Reply:
I've updated it to B.

Reply:
Yeah, I think you are more qualified to choose between A, B, and C so B it is!

Reply:
I thought of a D option. Let the user choose, and default to B?

```py
# options: "raise", "clamp", or "none" 
def subtract_distance_offset(self, distance_offset, negative_distance_handling="raise"):
````

You can still choose your favorite.

Reply:
I think I'm going to go for B for now, because I would like to know if users are using negative distances (and why) and it'd likely come up in support if there is a need for it.